test/CompletionExtSample/CompletionExt1.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices.Completion;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices.Completion.Extension;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices.Contracts;
+
+namespace CompletionExtSample
+{
+    [Export(typeof(ICompletionExtensionProvider))]
+    public class CompletionExtProvider1 : ICompletionExtensionProvider
+    {
+        Task<ICompletionExtension> ICompletionExtensionProvider.CreateAsync(IReadOnlyDictionary<string, object> properties, CancellationToken cancellationToken)
+        {
+            return Task.FromResult<ICompletionExtension>(new CompletionExt1(properties));
+        }
+    }
+
+    public class CompletionExt1 : ICompletionExtension
+    {
+        public string Name => "CompletionExt1";
+
+        private readonly string _modelPath;
+
+        public CompletionExt1(IReadOnlyDictionary<string, object> properties)
+        {
+            _modelPath = (string)properties["modelPath"];
+        }
+
+        void IDisposable.Dispose()
+        {
+            throw new NotImplementedException();
+        }
+
+        async Task<CompletionItem[]> ICompletionExtension.HandleCompletionAsync(ConnectionInfo connInfo, ScriptDocumentInfo scriptDocumentInfo, CompletionItem[] completions, CancellationToken token)
+        {
+            if (completions == null || completions == null || completions.Length == 0)
+            {
+                return completions;
+            }
+            await Run(completions, token);
+            return completions;
+        }
+
+        async Task ICompletionExtension.Initialize(CancellationToken token)
+        {
+            await LoadModel(token).ConfigureAwait(false);
+            return;
+        }
+
+        private async Task LoadModel(CancellationToken token)
+        {
+            //loading model logic here
+            await Task.Delay(2000).ConfigureAwait(false); //for testing
+            token.ThrowIfCancellationRequested();
+            Console.WriteLine("Model loaded from: " + _modelPath);       
+        }
+
+        private async Task Run(CompletionItem[] completions, CancellationToken token)
+        {
+            Console.WriteLine("Enter ExecuteAsync");
+
+            var sortedItems = completions.OrderBy(item => item.SortText);
+            sortedItems.First().PreSelect = true;
+            foreach(var item in sortedItems)
+            {
+                item.Command = new Command
+                {
+                    command = "vsintellicode.completionItemSelected",
+                    arguments = new object[] { new Dictionary<string, string> { { "IsCommit", "True" } } }
```

Comment:
So what is this command doing? Why do you need to call additional commands on the ADS side?

Reply:
This is the command exposed on the IntelliCode side for logging telemetry when one completion item is selected.